File: level1/level1-7/level1-7.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> d, int budget) {
	int answer = 0;

	sort(d.begin(), d.end());
	int cnt = 0;

	for (int i = 0; i < d.size(); i++) {
		budget -= d[i];
		cnt += 1;

		if (budget < 0) {
			cnt -= 1;
			budget += d[i];
			break;
		}
	}

	answer = cnt;

	return answer;
}

int main()
{
	solution(vector<int> { 1, 3, 2, 5, 4 }, 9);
	solution(vector<int> { 2, 2, 3, 3 }, 10);
}

File: level1/level1-7/level1-7.py
def solution(d, budget):
    answer = 0
    d.sort()
    print(d)
    cnt = 0

    for i in range(len(d)):
        budget -= d[i]
        cnt += 1

        if (budget < 0):
            cnt -= 1
            budget += d[i]
            break;

    answer = cnt
    return answer

# other solution
# def solution(d, budget):
#     d.sort()
#     while budget < sum(d):
#         d.pop()
#     return len(d)

if __name__ == "__main__":
    # execute only if run as a script
    solution([1,3,2,5,4], 9)
    solution([2,2,3,3], 10)